integration-tests/rest-tests-jaxrs/src/test/groovy/org/hawkular/metrics/rest/GaugesITest.groovy

```diff
@@ -466,7 +466,135 @@ Actual:   ${response.data}
           bucketPoint << [empty: true]
           break
       }
-      expectedData.push(bucketPoint);
+      expectedData.push(bucketPoint)
+    }
+
+    assertNumericBucketsEquals(expectedData, response.data ?: [])
+  }
+
+  @Test
+  void fetchStatsWithPercentiles() {
+    String gauge = "G1"
+
+    def response = hawkularMetrics.post(
+        path: "gauges/$gauge/raw",
+        headers: [(tenantHeaderName): tenantId],
+        body: [
+            [timestamp: 60_000 * 1.0, value: 321.8],
+            [timestamp: 60_000 * 7.0, value: 468.0],
+            [timestamp: 60_000 * 7.2, value: 472.0],
+            [timestamp: 60_000 * 7.5, value: 968.0],
+            [timestamp: 60_000 * 7.6, value: 474.0],
+        ]
+    )
+    assertEquals(200, response.status)
+
+    response = hawkularMetrics.get(
+        path: "gauges/$gauge/stats",
+        headers: [(tenantHeaderName): tenantId],
+        query: [start: 60_000, end: 60_000 * 8, buckets: 7, percentiles: '50,99']
+    )
+    assertEquals(200, response.status)
+    assertContentEncoding(response)
+
+    def expectedData = []
+    (1..7).each { i ->
+      Map bucketPoint = [start: 60_000 * i, end: 60_000 * (i + 1)]
+      double val
+      switch (i) {
+        case 1:
+          val = 321.8
+          bucketPoint << [min: val, avg: val, median: val, max: val, sum: val,
+                          percentiles: [
+                              [value: val, originalQuantile: 50, quantile: 50.0],
```

Comment:
Apart from originalQuantile leaking (I'm assuming lack of it in other integration-tests means it's not leaked there), I'd say this PR is fine. 

If it's leaking everywhere in the REST API, file a JIRA and lets remove it (it's not intended to be in the response)